pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -317,15 +320,20 @@ async def run(self):
                 # update jira promote task status
                 self._logger.info("Updating promote release subtask")
                 jira_issue_key = group_config.get("release_jira")
-                if jira_issue_key:
+                if jira_issue_key and not self.dry_run:
                     parent_jira = self._jira_client.get_issue(jira_issue_key)
-                    subtask = self._jira_client.get_issue(parent_jira.fields.subtasks[3].key)
-                    self._jira_client.add_comment(
-                        subtask,
-                        "promote release job : {}".format(os.environ.get("BUILD_URL"))
-                    )
-                    self._jira_client.assign_to_me(subtask)
-                    self._jira_client.close_task(subtask)
+                    title = "[Wed] Promote the tested nightly"
+                    subtask = next((s for s in parent_jira.fields.subtasks if title in s.fields.summary), None)
+                    if not subtask:
+                        raise ValueError("Promote release subtask not found in release_jira: %s", jira_issue_key)
```

Comment:
I think we can make this a warning since we are matching the subtask by title.

Reply:
I kept it so if someone changes the title of the subtask in template or this specific ticket we would get to know about it